### PR TITLE
disks: Don't take defaults from existing volumes when creating a new one

### PR DESCRIPTION
**Pull request by mvollmer**
Specifically, don't declare the new device for a new volume to be a CDROM when the first volume in the selected pool happens to be a ISO.

### CODE REVIEW

**Comment by mvollmer** [comment]
This was hella confusing. I have a Alpine ISO in my default pool, which meant that each newly created volume in that pool would be added as a CDROM to the machine. I didn't get to use a virtio bus, and the image format was always raw. Very magic.  I blame `useEffect`.

**Comment by mvollmer** [comment]
Yes, yes, a test would be nice... :-)

**Review by martinpitt — DISMISSED** [review]
Thanks!

**Comment by mvollmer** [comment]
> Yes, yes, a test would be nice... :-)

Added, and it found a bug. :-)

**Review by martinpitt — DISMISSED** [review]
Hah, nice!

**Comment by martinpitt** [comment]
Rebased, to get a new run after re-dropping cloud-init from alpine. The previous run was way too red.

**Comment by martinpitt** [comment]
This [breaks testAddDiskDirPool everywhere](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1617-1921153b-20240508-100425-arch/log.html#80), I'm afraid

**Comment by mvollmer** [comment]
> This [breaks testAddDiskDirPool everywhere](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1617-1921153b-20240508-100425-arch/log.html#80), I'm afraid

That was the switch to q35 by default... now cdroms can't be hotplugged anymore apparently (except on rhel-9???). I rearrange the tests a bit.

File: src/components/vm/disks/diskAdd.jsx [hunk 1]
```diff
@@ -397,7 +397,7 @@ export const AddDiskModalBody = ({ disk, idPrefix, isMediaInsertion, vm, vms, su
 
     const getPoolFormatAndDevice = (pool, volName) => {
         const params = { format: getDefaultVolumeFormat(pool), device: "disk" };
-        if (['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
+        if (volName && ['dir', 'fs', 'netfs', 'gluster', 'vstorage'].indexOf(pool.type) > -1) {
```
Comment (martinpitt) [inline src/components/vm/disks/diskAdd.jsx:400]: To make sure I understand: Is this functionally required to fix this bug, or just an unrelated optimization? I.e. are there pool volumes with undefined names which would unintentionally match here?

**Review by martinpitt — APPROVED** [review]
Danke!

**Pull request merged**